cnnimg.py

```diff
@@ -65,111 +62,107 @@ def imageprocessing(self, inputlocation, outputlocation, tempA, tempB, initialco
         u = np.array(gray)
         u = u[:,:,0]
         z0 = (u)*initialcondition
-        self.Bu = sig.convolve2d(u, tempB, 'same')
-        self.tempA = tempA
-        self.Ib = Ib
-        z0 = z0.reshape(z0.size)
-        z = self.cnn(sint.odeint(self.f, z0, t))
+        Bu = sig.convolve2d(u, tempB, 'same')
+        z0 = z0.flatten()
+        z = self.cnn(sint.odeint(self.f, z0, t, args=(Ib, Bu, tempA)))
         l = z[z.shape[0]-1,:].reshape((self.n, self.m))
         l = l/(255.0)
-        for i in range(l.shape[0]):
-            for j in range(l.shape[1]):
-                l[i][j] = np.uint8(round(l[i][j]*255))
+        l = np.uint8(np.round(l*255))
```

Comment:
I used manual loops for this operation, since the direct function was causing failure when tested on a Raspberry Pi (might have some incompatibility with Linux). The direct function works fine in Windows though.